### ISSUE TITLE
Example implementation of BYOL

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#1366]
## 🚀 Feature

It would be nice to provide [BYOL](https://arxiv.org/abs/2006.07733) implementation with Ignite. For reference, we can take this repository: https://github.com/lucidrains/byol-pytorch

**Comment by shngt** [comment]
Hi, is this still a desired feature?

**Comment by vfdev-5** [comment]
@shngt yes, it is still open.

**Comment by shngt** [comment]
The BYOL paper uses ImageNet for experiments, but it can't be downloaded via torchvision. Should I use another dataset for the example, or just implement the DataLoaders and ask the user to download the dataset on their own? Or is there some other protocol to be followed?

**Comment by sdesrozis** [comment]
It would be nice to have an ImageNet example. It should be added to reference use cases.

But maybe in a first step, we could provide an example using cifar which is available with torchvision. 

What do you think ?

**Comment by shngt** [comment]
I agree - using CIFAR would not help much with ensuring correctness of the implementation. Thank you for pointing this out this file :)

**Comment by sally20921** [comment]
Please note that I am in the process of implementing BYOL and BYOLv2(SimSiam) in ImageNet myself!
Please visit:
BYOL - https://github.com/sally20921/BYOL
BYOLv2 - https://github.com/sally20921/SimSiam

**Comment by sdesrozis** [comment]
> Please note that I am in the process of implementing BYOL and BYOLv2(SimSiam) in ImageNet using Ignite myself!
> Please visit:
> BYOL - https://github.com/sally20921/BYOL
> BYOLv2 - https://github.com/sally20921/SimSiam

@sally20921 Very excited to see your BYOL implementation powered by ignite ! I will have a look as soon as possible !
